### DISCUSSION TITLE
HA unsupported device - External converter causes constant restart with no log messages

### DISCUSSION THREAD

**Discussion by vanceb** [Koenkk/zigbee2mqtt#11897] [Devices]
I am running:

* Home Assistant 2022.3.4
* Zigbee2mqtt add-on installed from the official store (1.18.1-1)

I am trying to add support for an innr bulb BY 266.  I have followed the guide for using an external converter and my `by266.js` file is as follows:

~~~js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['BY 266'],
    model: 'BY 266',
    vendor: 'innr',
    description: 'Bayonet bulb with dimming',
    extend: extend.light_onoff_brightness(),
};

module.exports = definition;
~~~

This file is placed in the same directory as the zigbee2mqtt `configuration.yaml`.

I then edit the zigbee2mqtt config using the web UI  Config -> External Converters and add `by266.js` and set the log level to `debug`.  When I click the restart button zigbee2mqtt restarts but never comes up.  Looking at the logs directory a new subfolder is created about every 5 seconds and the `log.txt` file in the subdirectory is empty.

Restarting the add-on using HA Supervisor reverts the config file and the addon then restarts, but has obviously now not loaded my external converter.  I am struggling what to try next.

Grateful for any help in getting this to work.

**Comment by Koenkk** [comment]
You are using an old z2m version: 
If you are still on 1.18.1 check first point of https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.19.0

**Reply by vanceb** [reply]
OK, I have changed the repository that zigbee2mqtt is downloaded from to https://github.com/zigbee2mqtt/hassio-zigbee2mqtt, but unfortunately I managed to "forget" all of my devices and am having to go around and re-pair them.

On the positive note I am now able to get the external converter to be recognised, but points to note:

1. The location of the zigbee2mqtt directory has changed from `/root/share/zigbee2mqtt` to `/root/config/zigbee2mqtt`, so the external converter must be placed in the new location
2. Updating the config through the zigbee2mqtt UI doesn't seem to stick - You have to modify the config file using the HA Supervisor UI to be able to add an external converter, otherwise this config change is forgotten when the add-on is restarted by the supervisor.
